lib/internal/http2/compat.js

```diff
@@ -308,8 +306,12 @@ class Http2ServerRequest extends Readable {
   _read(nread) {
     const state = this[kState];
     if (!state.closed) {
-      state.didRead = true;
-      process.nextTick(resumeStream, this[kStream]);
+      if (!state.didRead) {
+        state.didRead = true;
+        this[kStream].on('data', onStreamData);
+      } else {
+        process.nextTick(resumeStream, this[kStream]);
```

Comment:
That part is carried over from before (didn't change) and used to definitely be necessary. We have a test for it even which breaks otherwise: `parallel/test-http2-compat-serverrequest-pipe`.

Reply:
You could probably have a look at the git blame for the line to find the corresponding PR.

Reply:
See https://github.com/nodejs/node/pull/15503 — I don't fully recall where exactly we have an async call to pause but we do somewhere.

Reply:
It's probably something we would have to look into this :D.